### PR TITLE
Improvements in ASCollectionGalleryLayoutDelegate

### PR DESCRIPTION
**Pull request by nguyenhuy**
- `ASCollectionGalleryLayoutDelegate` now can handle section inset, as well as interitem and line spacings (thanks to #472).
- Other small changes.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please ensure license is correct for ASCollectionGalleryLayoutDelegate.mm:</p>
<pre><code>//
//  ASCollectionGalleryLayoutDelegate.mm
//  Texture
//
//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
</code></</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please ensure license is correct for _ASCollectionGalleryLayoutInfo.m:</p>
<pre><code>//
//  _ASCollectionGalleryLayoutInfo.m
//  Texture
//
//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
</code></</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/4571f5c13901f1b893411a960cacdf90/raw/4c30c924bf29332f61b6c76ee95170fc0ebcdb64/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Source/Details/ASCollectionGalleryLayoutDelegate.mm [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    unsigned int sectionInsetForElements:1;
+  } _propertiesProviderFlags;
+}
+
+- (instancetype)initWithScrollableDirections:(ASScrollDirection)scrollableDirections
+{
+  self = [super init];
+  if (self) {
+    // Scrollable directions must be either vertical or horizontal and can't be both
+    ASDisplayNodeAssertFalse(ASScrollDirectionContainsVerticalDirection(scrollableDirections)
```
Comment (garrettmoon) [inline Source/Details/ASCollectionGalleryLayoutDelegate.mm]: Should we assert that it has at least one scrollable direction?

Comment (nguyenhuy) [inline Source/Details/ASCollectionGalleryLayoutDelegate.mm]: Good point. I'll add it. Thanks for the review.

**Pull request merged**